mobile/apps/photos/plugins/ente_feature_flag/lib/src/service.dart

```diff
@@ -42,7 +42,17 @@ class FlagService {
 
   bool get disableCFWorker => flags.disableCFWorker;
 
-  bool get internalUser => flags.internalUser || kDebugMode;
+  /// Returns true if the user is an internal user, respecting the debug toggle.
+  /// Use [isInternalUserOrDebug] for UI visibility that should persist even when disabled.
+  bool get internalUser {
+    final isDisabled = _prefs.getBool("ls.internal_user_disabled") ?? false;
+    return (flags.internalUser || kDebugMode) && !isDisabled;
+  }
+
+  /// Returns true if the user is an internal user OR in debug mode,
+  /// ignoring the debug toggle. Use this for showing debug UI elements.
+  bool get isInternalUserOrDebug => flags.internalUser || kDebugMode;
```

Comment:
We can remove this and just use flags.internalUser || kDebugMode; for debug section?